packages/client/tests/functional/_utils/setupQueryEngine.ts

```diff
@@ -12,31 +9,29 @@ import lockfile from 'proper-lockfile'
  * normally the downloading of the required engine is done in `getGenerators`. As the test
  * clients bypass this we need to ensure the correct engine is present.
  * @param clientEngineType
- * @param platform
  */
-export async function setupQueryEngine(clientEngineType: ClientEngineType, platform: Platform) {
+export async function setupQueryEngine(clientEngineType: ClientEngineType) {
```

Comment:
If I read this correctly, after this change every test suite will always download an engine during `beforeAll`? If so, we might bring flakiness back that way: lock lasts only for duration of `setupQueryEngine`. So, previously, when engine did not exist, first test suite acquired the lock, downloaded the engine and continued. All subsequent workers will now immediately realize that there is already an engine file and skip the download. After this change, following situation is possible:
- Worker A acquires lock, downloads engine, releases the lock and proceeds with the tests
- Worker B acquires the lock, starts download, in process removes the engine file, which is still used by Worker A.

I could see a couple of solutions for this:
- keep existence check, but make it more sofisticated, i.e.  check checksum of the existing engine file or compare existing and desired engine
- move engine setup out of jest completely and do it in `helpers/functional-tests` before running any tests instead.
- keep simple existence check, but remove all downloaded engines in `bump-engines` script.

Reply:
Thanks @SevInf, I have applied the suggested changes to use the version as an invalidation method.